### PR TITLE
gnome-photos: 3.32.0 → 3.32.1

### PR DESCRIPTION
**Pull request by jtojnar**
* Moves some GNOME packages to top-level namespace
* Upgrades gnome-photos
* Adds installed tests

### CODE REVIEW

**Comment by worldofpeace** [comment]
@jtojnar I think `gnome-photos` tests probably needs dconf.

**Comment by jtojnar** [comment]
Yeah, but it is not enough. Currently debugging that.

**Comment by jtojnar** [comment]
With the following debug patch

```diff
diff --git a/nixos/modules/services/desktops/gnome3/at-spi2-core.nix b/nixos/modules/services/desktops/gnome3/at-spi2-core.nix
index cca98c43dc7..6b3b3f7d0ad 100644
--- a/nixos/modules/services/desktops/gnome3/at-spi2-core.nix
+++ b/nixos/modules/services/desktops/gnome3/at-spi2-core.nix
@@ -4,7 +4,17 @@
 
 with lib;
 
-{
+let
+  package = pkgs.at-spi2-core.overrideAttrs (attrs: {
+    nativeBuildInputs = attrs.nativeBuildInputs ++ [ pkgs.makeWrapper ];
+
+    postFixup = ''
+      wrapProgram $out/libexec/at-spi-bus-launcher \
+        --prefix XDG_DATA_DIRS : ${pkgs.gsettings-desktop-schemas}/share/gsettings-schemas/${pkgs.gsettings-desktop-schemas.name}
+    '';
+  });
+
+in {
 
   ###### interface
 
@@ -30,9 +40,9 @@
 
   config = mkMerge [
     (mkIf config.services.gnome3.at-spi2-core.enable {
-      environment.systemPackages = [ pkgs.at-spi2-core ];
-      services.dbus.packages = [ pkgs.at-spi2-core ];
-      systemd.packages = [ pkgs.at-spi2-core ];
+      environment.systemPackages = [ package ];
+      services.dbus.packages = [ package ];
+      systemd.packages = [ package ];
     })
 
     (mkIf (!config.services.gnome3.at-spi2-core.enable) {
diff --git a/nixos/tests/gnome-photos.nix b/nixos/tests/gnome-photos.nix
index 74c1aea4db2..aa5c4d8def0 100644
--- a/nixos/tests/gnome-photos.nix
+++ b/nixos/tests/gnome-photos.nix
@@ -16,6 +16,7 @@
 
   testScript = ''
     $machine->waitForX;
-    $machine->succeed("gnome-desktop-testing-runner -d '${pkgs.gnome-photos.installedTests}/share'");
+    $machine->execute("env G_MESSAGES_DEBUG=all gnome-desktop-testing-runner -d '${pkgs.gnome-photos.installedTests}/share' gnome-photos/basic.py.test");
+    $machine->execute("cat /tmp/dogtail-root/logs/.basic-wrapped_*_debug");
   '';
 })

```

all warnings disappear but the test is still failing:

```
machine: running command: env G_MESSAGES_DEBUG=all gnome-desktop-testing-runner -d '/nix/store/0vw0sgclpn171hnnivvwgwpsgsdms6y5-gnome-photos-3.32.1-installedTests/share' gnome-photos/basic.py.test
machine# [   26.102431] dbus-daemon[803]: [session uid=0 pid=801] Activating service name='org.a11y.Bus' requested by ':1.0' (uid=0 pid=839 comm="/nix/store/2zqdjd1xrg4nljkkxi9mw6bgy132rz1x-python")
machine# [   26.194530] dbus-daemon[803]: [session uid=0 pid=801] Successfully activated service 'org.a11y.Bus'
machine# [   26.238909] org.a11y.Bus[803]: dbus-daemon[855]: Activating service name='org.a11y.atspi.Registry' requested by ':1.0' (uid=0 pid=839 comm="/nix/store/2zqdjd1xrg4nljkkxi9mw6bgy132rz1x-python")
machine# [   26.423570] org.a11y.Bus[803]: dbus-daemon[855]: Successfully activated service 'org.a11y.atspi.Registry'
machine# [   26.438842] org.a11y.Bus[803]: SpiRegistry daemon is running with well-known name - org.a11y.atspi.Registry
machine# [   26.591853] dbus-daemon[803]: [session uid=0 pid=801] Activating service name='ca.desrt.dconf' requested by ':1.3' (uid=0 pid=839 comm="/nix/store/2zqdjd1xrg4nljkkxi9mw6bgy132rz1x-python")
machine# [   26.695121] dbus-daemon[803]: [session uid=0 pid=801] Successfully activated service 'ca.desrt.dconf'
machine# [   27.006613] gnome-desktop-testing-runner[837]: Executing: gnome-photos/basic.py.test
machine# [   27.590633] gnome-desktop-testing-runner[837]: FAIL: gnome-photos/basic.py.test (Child process exited with code 1)
machine# [   27.613062] gnome-desktop-testing-runner[837]: SUMMARY: total=1; passed=0; skipped=0; failed=1; user=0.7s; system=0.5s; maxrss=67848
```

It succeeds in my user session:

`nix run -f . gnome-desktop-testing ostree -c env G_MESSAGES_DEBUG=all gnome-desktop-testing-runner -d (nix-build -A gnome-photos.installedTests --no-out-link)/share gnome-photos/basic.py.test`

**Comment by jtojnar** [comment]
The missing part was adding `gnome-photos` to `services.dbus.packages` and enabling the accessibility for dogtail to work. Wow, that is hideous – perhaps we should create a package providing `gsettings` wrapped with desktop schemas.

**Comment by worldofpeace** [comment]
I came upon that as well maybe 15 minutes ago. It caused an involuntary cringe, I had to look away :rofl:

**Comment by worldofpeace** [comment]
>  Wow, that is hideous – perhaps we should create a package providing gsettings wrapped with desktop schemas.

Is there a metric on how many other tests use dogtail? I'd dislike having to copy the same hack.

**Comment by jtojnar** [comment]
I think it is quite rare. Some packages were even removing it as it has not been maintained until recently.

File: nixos/tests/gnome-photos.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    programs.dconf.enable = true;
+    services.gnome3.at-spi2-core.enable = true; # needed for dogtail
+    environment.systemPackages = with pkgs; [ gnome-desktop-testing ];
+    services.dbus.packages = with pkgs; [ gnome-photos ];
+  };
+
+  testScript = ''
+    $machine->waitForX;
+    # dogtail needs accessibility enabled
+    $machine->succeed("env GIO_EXTRA_MODULES='${lib.getLib pkgs.gnome3.dconf}/lib/gio/modules' XDG_DATA_DIRS='${pkgs.gsettings-desktop-schemas}/share/gsettings-schemas/${pkgs.gsettings-desktop-schemas.name}' ${pkgs.glib.bin}/bin/gsettings set org.gnome.desktop.interface toolkit-accessibility true 2>&1");
```
Comment (worldofpeace) [inline nixos/tests/gnome-photos.nix]: Can you make `let` the environment variables string into the `testScript` so it's clearer to read?

**Review by worldofpeace — APPROVED** [review]
LGTM with above comment.

**Comment by worldofpeace** [comment]
Oh, I see warnings in the test for tracker dbus interfaces
```
machine# [   20.071667] .gnome-photos-w[903]: Unable to create PhotosTrackerQueue: Failed to load SPARQL backend: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Tracker1 was not provided by any .service files
machine# [   20.131247] .gnome-photos-w[903]: Unable to create PhotosTrackerQueue: Failed to load SPARQL backend: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Tracker1 was not provided by any .service files
```

Think we should enable them?

**Comment by jtojnar** [comment]
I tried but then it also wants dleyna and since the test just checks if the interface works, I did not want to descend into this rabbit hole.

**Review by worldofpeace — APPROVED** [review]
LGTM. No :rabbit2: today :smile:

**Pull request merged**